### ISSUE TITLE
classify: resolve arm offsets by counting query bases, per the bundle's features.resolution

### ISSUE THREAD

**Issue by jayhesselberth** [rnabioco/escapepod-rs#212]
## Summary

The charging corpus now resolves common-arm feature offsets by **counting query
bases from the junction**, not by looking them up in the alignment.
`crates/escapepod-classify/src/anchor.rs` does the latter, so as of
escapepod-models `9a37d46` escpod computes different features from the ones the
shipped weights were trained on — and it differs on exactly the reads that
matter.

## Why the change was made

bwa's alignment stops at the aminoacyl adduct. With the alignment-derived rule,
whether an arm offset had features depended on how far bwa got, and that
depended on charging: **20.9–25.3% of charged-library reads lost the entire arm
against 0.9% of uncharged**. The model read that missing pattern as a class.

Counting never consults the aligner past the junction. Soft-clipped bases still
carry move-table spans (92.8% of affected reads have ≥16 query bases there,
median 58), so the walk works past the end of the alignment and every read gets
the same features.

## What escpod needs to do

For arm offsets `o` in `1..=N`, the query position is **`q(junction) + o`** —
walked along the query. Offsets above `N` are unresolved (NaN) for every read.
Body offsets (`o <= 0`) keep the exact-alignment rule, no slop backfill.

The bundle declares all of it, so this should be read from `metadata.json`
rather than hardcoded:

```json
"features": {
  "resolution": {
    "arm_offsets": "count",
    "count_arm_bases": 8,
    "body_offsets": "exact alignment only (no slop backfill)"
  }
}
```

`signal.mask` moves with it: the boundary is the first raw sample of the base
at `q(junction) + count_arm_bases`, **not** the aligner's `q(motif+3+16)`,
which is unavailable on precisely the reads this fixes.

## Why N is capped at 8, and why that cap is load-bearing

The basecall indels through the disrupted arm, so counting too far steps past
the arm into the divergent 13-mer — which is the library label. Measured on 40k
reads per library:

| N | leak, charged | leak, uncharged | asymmetry |
|---|---|---|---|
| 8 | 0.33% | 0.05% | +0.003 |
| 12 | 3.23% | 1.44% | +0.018 |
| 16 | 43.5% | 27.5% | +0.160 |

Do not raise N without re-measuring; the asymmetry is what matters, not the
rate. `scripts/charging/count_anchor.py` in escapepod-models reproduces it.

## Effect

Paired over 5 seeds on an identical 195,189-read test set: AUROC 0.9583 →
0.9578 (−0.0005, **sign flips across seeds**, i.e. not an effect) and balanced
accuracy 0.8725 → **0.9052** (+0.0327, same sign every seed). At the shipped
hyperparameters the GBM goes 0.9614/0.8736 → **0.9629/0.9117**.

## Ordering

This sits behind #211 (the backfill leak and the fabricated features for
unaligned offsets) — that is a correctness bug in shipped code and should land
first. #210 (abstain) has a corrected rule; see the comment there.

**Comment by jayhesselberth** [comment]
Implemented. `SpanMode::from_arm_bases(features.resolution.count_arm_bases)` selects the mode from the bundle rather than a flag (`anchor.rs:399`), and arm offsets `o >= 1` resolve by walking the query from the junction (`anchor.rs:475`). Soft-clipped bases still carry move-table spans, so it works past the end of the alignment. A bundle without the block gets the aligner path, which is what the pre-2026-08-10 models genuinely used.

Pinned by a golden test (`counted_anchor_matches_reference`) generated from the current reference implementation.

Follow-up that came out of it: #226 fixed the counting mode's fallback, which went straight to the junction where the reference falls back to any resolved arm base first — found by comparing a real corpus, since all 19 fixture reads resolve as `counted` and never exercise the fallback.
